### PR TITLE
introduce class for description block splitting

### PR DESCRIPTION
**Pull request by stijnvermeeren-swisstopo**
Object-oriented approach to the different methods for partitioning description lines into blocks.

### CODE REVIEW

**Comment by stijnvermeeren-swisstopo** [comment]
@redur was meinst du zu diese Code-Struktur?

File: src/stratigraphy/util/find_description.py [hunk 1]
```diff
@@ -112,145 +110,8 @@ def get_description_blocks(
         # indicators are dropped. That's why we have to consider vertical spacing as well.
         _blocks = []
         for block in blocks:
-            _blocks.extend(_split_block_by_vertical_spacing(block.lines, threshold=threshold))
+            _blocks.extend(splitter.create_blocks(block.lines))
```
Comment (redur) [inline src/stratigraphy/util/find_description.py:113]: Nice that we can reuse the same object here.

**Review by redur — APPROVED** [review]
Thank you Stijn. I believe you are right and this is overall more readable and better abstracted than it was before.

Did you check if executing before and after leads to the same results?

**Comment by stijnvermeeren-swisstopo** [comment]
I verified that the scores are identical for all documents, between the current main branch and this branch. (Overall F1 score: 86.6%.)

**Pull request merged**